### ISSUE TITLE
Let `id` and `nick` Properties return nil on non-network-components instead of an thrown exception

### ISSUE THREAD

**Issue by Feuermurmel** [Panakotta00/FicsIt-Networks#359]
**Describe the bug**

This is more of an enhancement request than a bug report.

Some actors don't have an `id` or `nick` property, probably because they can't be connected to using a network cable. Accessing these properties leads to an error. In some cases, it would be nice if those properties always existed and instead returned `nil` on those actors.

**To Reproduce**

Connect a constructor with its Nick set to `The Constructor` to the computer and add conveyor belts on both ends of the constructor. The run the following code:

```lua
local components = component.findComponent("The Constructor")
local component = component.proxy(components[1])

print(component.id, component.nick)

local nextComponent = component:getFactoryConnectors()[1]:getConnected().owner

print(nextComponent)
print(nextComponent.id)
print(nextComponent.nick)
```

The constructor as an ID and a Nick, which is printed. `component:getFactoryConnectors()[1]:getConnected().owner` will get a reference to one of the conveyor belts attached to the constructor, accessing `id` or `nick` will lead to the following error:

```
2024.11.17-14.28.47 [Info] 6F6C1E194301152DCA889F9EC34099E9 The Constructor
2024.11.17-14.28.47 [Info] Object<Build_ConveyorBeltMk3_C>
2024.11.17-14.28.47 [Fatal] id
                            stack traceback:
                            	[C]: in metamethod 'index'
                            	boot.lua:9: in main chunk
                            	[C]: in field 'doFile'
                            	EEPROM:4: in main chunk
```

**Expected behavior**

I'm not saying this behavior is wrong, but it is very surprising because the API is object oriented, with may things being objects that have a certain class. In other object-oriented environments, the set of properties and methods available on an object is usually the same between every object of the same class (e.g. in Java or C++ etc.).

Having these properties always exist and return `nil` would make some code simpler, because then value of the property could be checked using `==` instead of having to wrap the code in `pcall()` or similar.

**Additional context**

I'm using FicsIt Networks 0.3.27 and Satisfactory 1.0.0.6.

**Comment by RozeDoyanawa** [comment]
Java would give you compile time errors when trying to access properties that does not exist. Unless you did reflection magic and captured errors and turned them to runtime exceptions. 

Whilst I did not agree with the change to error on accessing "missing" properties, it is logical. 
Also, nick and id is magical properties. They only exist because lua runtime "captures" their access and returns a value if valid, so in theory, they do not belong to any class.

Imo, with the current idea of throwing errors on accessing undeclared properties/members, I think its logical to do the same for id and nick. However, perhaps pana could add a component-function (ex component.isNetworkComponent(obj)) to check if a component can be network connected.

**Comment by Panakotta00** [comment]
@Feuermurmel
Well, modern Object Oriented Programming...
1. usually provides interfaces or multiple inheritance. Those ids and nicks are essentially interfaces, but our custom reflection system doesnt have interfaces implemented, the unreal reflection system however does. Before the Lua Reflection and Module update id and nick behaved like Roze said, they were not even part of the Reflection system, instead they just "appeared out of no where" in the case the object was an network component. But after a poll on the discord it changed to be properties of the Object Class so the id and nick properties finally get documented properly (especially ingame using the reflection viewer).
And even if I would have implemented interfaces into the FIN Reflection System, for the Lua code nothing really would have changed I guess, the properties would not exist.
2. OOP doesnt have to use Inheritance, modern approaches go with composition instead. Tho some argue its not OOP then anymore. But ngl, nobody knows what OOP actually is and different people have different opinions on it.

also @RozeDoyanawa There is already a [`isNetworkComponent` property](https://docs.ficsit.app/ficsit-networks/latest/reflection/classes/Object.html#_isnetworkcomponent_bool).

**Comment by RozeDoyanawa** [comment]
That property is pretty pointless since calling it on a object that does not have it still gives an error. I think the point OP wanted was something to tell IF an arbitrary thing is a network thing. And it seems more logical to put it in the component api so it can always return true or false and never error

**Comment by Panakotta00** [comment]
@RozeDoyanawa Every object has that property tho.

**Comment by Feuermurmel** [comment]
> Java would give you compile time errors when trying to access properties that does not exist. Unless you did reflection magic and captured errors and turned them to runtime exceptions.

Yes, but at least per the documentation, the properties are declared on `Object`, so even Java would not give a compile time error in that case.

I mentioned Java as an example because I know that "OOP" is not really a term with a fixed meaning, so I wanted to use an example that I know well and that somewhat matches the concepts of FicsIt Networks. I could have also used Python as an example, where invalid property accesses are not checked at compile time and instead at runtime. But I think Python is still very much an object-oriented programming language.

---

In the meantime, I did find the `isNetworkComponent` property and figured out that it could be used to check whether accessing the `id` and `nick` properties is possible.

I think I would have figured this out quicker if the documentation of `id` and `nick` mentioned that access is only possible if `isNetworkComponent` returns true.

I interpreted the wording "Only available for Network Components" at that the properties always exist, but that the information is not available in some cases. IMHO it is very common to have properties with information that is only available sometimes to contain `nil` when the information is not available.

Also, in Lua, it is usually possible to use `obj.prop ~= nil` to check whether a property exists in a table. But I know that the proxy object doesn't behave like that and instead gives an error on unknown properties, which I think is a good idea.
